boards/common/stm32/Kconfig.l4

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2020 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menu "STM32 L4/WB clock configuration"
+    depends on CPU_FAM_L4 || CPU_FAM_WB
+
+choice
+bool "Clock source selection"
+default USE_CLOCK_PLL
+
+config USE_CLOCK_PLL
+    bool "Select PLL as system clock"
+
+config USE_CLOCK_MSI
+    bool "Select MSI as system clock"
+
+config USE_CLOCK_HSE
+    bool "Select HSE (High Speed External) oscillator as system clock" if BOARD_HAS_HSE
+
+config USE_CLOCK_HSI
+    bool "Select HSI16 (High Speed Internal) oscillator as system clock"
+
+endchoice
+
+choice
+bool "Source clock for PLL" if USE_CLOCK_PLL
```

Comment:
Sorry, on my suggestion I added the prompt conditional to the choice, but this does not behave like `int` symbols with multiple defaults. Here the default options are not selected when the choice is not visible.